### PR TITLE
TiFlash add metrics port

### PR DESCRIPTION
**Pull request by mikechengwei**
### What problem does this PR solve?
Reslove #3184
### What is changed and how does it work?
TiFlash add metrics port.

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

**Review by csuzhangxc — APPROVED** [review]

File: pkg/manager/member/tiflash_member_manager.go [hunk 1]
```diff
@@ -281,6 +281,12 @@ func getNewHeadlessService(tc *v1alpha1.TidbCluster) *corev1.Service {
 					TargetPort: intstr.FromInt(int(20170)),
 					Protocol:   corev1.ProtocolTCP,
 				},
+				{
```
Comment (DanielZhangQD) [inline pkg/manager/member/tiflash_member_manager.go:284]: There are two metrics ports for TiFlash metrics, refer to https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/tiflash_member_manager.go#L448

Comment (mikechengwei) [reply pkg/manager/member/tiflash_member_manager.go:284]: done

File: pkg/manager/member/tiflash_member_manager.go [hunk 2]
```diff
@@ -281,6 +281,19 @@ func getNewHeadlessService(tc *v1alpha1.TidbCluster) *corev1.Service {
...
+					Name:       "metrics",
+					Port:       8234,
+					TargetPort: intstr.FromInt(int(8234)),
+					Protocol:   corev1.ProtocolTCP,
+				},
+
+				{
+					Name:       "tiflash-proxy",
```
Comment (DanielZhangQD) [inline pkg/manager/member/tiflash_member_manager.go:292]:
```suggestion
					Name:       "proxy-metrics",
```

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Comment by DanielZhangQD** [comment]
@mikechengwei Please also help add the yaml examples of the service monitors.

**Pull request merged**